backend/app/adapter/graphql/resolver/authquery.go

```diff
@@ -3,15 +3,20 @@ package resolver
 import (
 	"time"
 
+	"github.com/short-d/app/fw"
+
 	"github.com/short-d/short/app/adapter/graphql/scalar"
 	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/changelog"
 	"github.com/short-d/short/app/usecase/url"
 )
 
 // AuthQuery represents GraphQL query resolver that acts differently based
 // on the identify of the user
 type AuthQuery struct {
 	user         *entity.User
+	timer        fw.Timer
```

Comment:
Timer should be a dependency of `ChangeLog` instead of `AuthQuery`. Retrieving current time is a piece of core business logic, which we call an use case.  Moving timer into `ChangeLog` allows us to reuse `ChangeLog` use case without worrying about internal implementation. This also make `AuthQuery` easier to understand by removing the unrelated complexities.

Reply:
But getting lastViewedAt should be part of use case of userChangeLog which we have not implemented yet

Reply:
@kapillamba4 Then let’s add a fake API to changelog use case and make it work in the next PR!